### PR TITLE
Adds websocket versioning

### LINKED ISSUE #1471: Version access to the websocket API
Consider the following scenario:

1. A user installs eigen.
1. We make a breaking change to the way eigen communicates with the live integration websocket.
1. We release an update to eigen, but the user doesn't upgrade.
1. They try to access live on their outdated eigen install, but they're presented an inconsistent or misleading user interface.

This is a problem I'd like to mitigate. This is different from the usual REST APIs we use because websockets operate in a fundamentally different way (for example, we can always add new things to an endpoint response, but if we change the order of events sent form the websocket, or add a new event, we have new way to reflect that in eigen without updating the app).

My initial idea, discussed with @acjay, was have some sort of version compiled into the app and reflected on echo. If the app is "out of date", we fall back to the internal web browser for live auctions. @devangt has raised the idea of asking the user to update. I think this warrants some UX thought from @katarinabatina before we pursue an implementation.

### PR DESCRIPTION
**Pull request by ashfurrow**
There was discussion on #1471 about a more complex implementation where we would ask the user to update. That would ultimately be nice, but this is an minimum viable implementation that will do for now. 

Fixes #1471.

### CODE REVIEW

**Comment by orta** [comment]
looks good to me

**Comment by ashfurrow** [comment]
🙇

File: Artsy/App/ARSwitchBoard.m [hunk 1]
```diff
@@ -204,7 +207,7 @@ - (void)updateRoutes
     Route *route = self.echo.routes[@"ARLiveAuctionsURLDomain"];
     if (route) {
         id _Nullable (^presentNativeAuctionsViewControllerBlock)(NSURL *_Nonnull);
-        if ([AROptions boolForOption:AROptionsEnableNativeLiveAuctions]) {
+        if ([AROptions boolForOption:AROptionsEnableNativeLiveAuctions] && [self requiresUpdateForWebSocketVersionUpdate] == NO) {
```
Comment (ashfurrow) [inline Artsy/App/ARSwitchBoard.m]: When the `AROptions` check here goes away, when we ship for realz, then we can just leave the rest of the if expression in place.

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
        
            1 Warning
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
    
    
  </tbody>
</table>
  

  



Time | Class | Function |
| --- | ----- | ----- |244.9ms | [AuctionViewController.swift](/artsy/eigen/tree/ashfurrow-max-bids/Artsy/View_Controllers/Auction/AuctionViewController.swift#L181) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
204.1ms | [SwiftyJSON.swift](/artsy/eigen/tree/ashfurrow-max-bids/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
193.3ms | [WebSocket.swift](/artsy/eigen/tree/ashfurrow-max-bids/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
81.5ms | [Observable.swift](/artsy/eigen/tree/ashfurrow-max-bids/Pods/Interstellar/Sources/Observable.swift#L71) | private final func nextTokenHash() -> Int
68.6ms | [AuctionLotMetadataStackScrollView.swift](/artsy/eigen/tree/ashfurrow-max-bids/Artsy/View_Controllers/Live_Auctions/Views/AuctionLotMetadataStackScrollView.swift#L18) | required init(viewModel: LiveAuctionLotViewModelType, sideMargin: String)
60.5ms | [RefinementOptionsViewController+Private.swift](/artsy/eigen/tree/ashfurrow-max-bids/Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift#L63) | func stackView() -> ORStackView




<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

**Comment by ashfurrow** [comment]
🍏

**Pull request merged**